src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,75 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);

```

Comment:
`save` is deprecated
```suggestion
            session.persist(shoppingCart);
```

Reply:
I used save() according to checklist. 
For saving ShoppingCart use save() method instead of persist(). The reason is that ShoppingCart object will have the transient state before the session opened, but its User object will be detached. persist() method throws PersistenceException if working with detached entities, when save() returns it to managed state.